modules/gapi/src/executor/gstreamingexecutor.cpp

```diff
@@ -84,11 +84,19 @@ struct DataQueue {
         }
         q = std::move(ptr);
     }
-    explicit DataQueue(tag t)
-        : q(new cv::gimpl::stream::DesyncQueue()) {
-        GAPI_Assert(t == DESYNC);
-    }
 
+    explicit DataQueue(tag t) {
+        if (t == DESYNC) {
+            q.reset(new cv::gimpl::stream::DesyncQueue());
+        } else {
+            GAPI_Assert(t == DROP);
+            q.reset(new cv::gimpl::stream::DesyncQueue(
+                    [](const Cmd &cmd) {
+                        // NB: Need to filter "data" messages from others.
+                        return cv::util::holds_alternative<cv::GRunArg>(cmd);
+                    }));
```

Comment:
What's the reason to have this lambda here?

Reply:
It is completely unclear for a reader (and for the user of this code)
1) Why he should write code like this
2) What he should implement in this lambda.

Reply:
Added comment